libslide/src/evaluator_rules.rs

```diff
@@ -17,11 +17,40 @@ static DEFAULT_RULESET: &[&str] = &[
     "$a + 0 -> $a", // VariablePlusZero
 ];
 
+/// A set of expression-mapping rules.
+///
+/// Each rule is of the string form
+///
+///   "<expr> -> <expr>"
+///
+/// Where <expr> is any expression pattern. An expression pattern is similar to any other
+/// expression, differing only in its pattern matching variables. The form of pattern matching
+/// variables and the expressions they match are as follows:
+///   
+///   | pattern | matches        |
+///   |:------- |:-------------- |
+///   | _<name> | Any expression |
+///   | #<name> | A constant     |
+///   | $<name> | A variable     |
+///   | <name>  | A variable     |
+///
```

Comment:
why do we have $ if its a variable even without it? line 17 in variables.rs implies the $ isn't necessary. I may be missing a reasoning for this.

Reply:
I would like for just `$name` to be able to parse a variable. I'll change it, though it will require some changes to our parser -- that is why I allow `name` to be a variable too, because by default the parser will parse it as a var as well.

Reply:
ok, dont' worry about it on this PR. Add a todo

Reply:
Added